ddprof-test/src/test/java/com/datadoghq/profiler/AbstractProcessProfilerTest.java

```diff
@@ -4,15 +4,38 @@
 import java.io.IOException;
 import java.io.InputStreamReader;
 import java.util.ArrayList;
+import java.util.Collections;
 import java.util.List;
+import java.util.Map;
 import java.util.concurrent.TimeUnit;
+import java.util.concurrent.locks.LockSupport;
 import java.util.function.Function;
 
 import static org.junit.jupiter.api.Assertions.*;
 
 
 public abstract class AbstractProcessProfilerTest {
-    protected final boolean launch(String target, List<String> jvmArgs, String commands, Function<String, Boolean> onStdoutLine, Function<String, Boolean> onStderrLine) throws Exception {
+    public static final class LaunchResult {
```

Comment:
Nice to see we are investing in controlled behaviour tests.
If we want to extend these, we could discuss what would be a good structure.
Nothing blocking though it is hard to see the structure. I would expect something like:
```
TestProcessController controller = new TestProcessController(WorkloadType.CPU_INTENSIVE);
controller.launch();
controller.waitForReady();
controller.sendStartSignal();
controller.waitForWorkingOutput();
controller.assertProfilerDidX();
controller.shutdown();
```
I think we basically have different parts
- controller
- assertions
- workload
- protocol